### ISSUE TITLE
[RFE] provide SHA256 checksum 

### ISSUE THREAD

**Issue by knumskull** [sosreport/sos#1204]
The default checksum for sosreport is `md5sum`. On some support-portals, (e.g. Red Hat), the default checksum after uploading the file is `sha256sum`.
This would require an extra step for verification the successful upload. 

An additional `sha256sum` beside the already existing `md5sum` would be great. If generating to much overhead, defining the default checksum-method in `sos.conf` would be sufficient instead.

**Comment by pmoravec** [comment]
I understand MD5 checksum is so legacy feature we shall not replace until very solid rationale behind. *additional* checksum is an option, but:
- that might confuse users what checksum(s) file are relevant or why there are two
- usability of that file on a customer portal relies on customers to really upload the extra file (how many times a customer uploads the md5 file?)

Note that when using default "xz" compressing algorithm, it is calculating crc64 checksum of the archive by default - see man pages:

       -C check, --check=check
              Specify  the  type of the integrity check.  The check is calculated from the uncompressed data and stored in the .xz file.  This option has an effect only when compressing
              into the .xz format; the .lzma format doesn't support integrity checks.  The integrity check (if any) is verified when the .xz file is decompressed.

              Supported check types:

              none   Don't calculate an integrity check at all.  This is usually a bad idea.  This can be useful when integrity of the data is verified by other means anyway.

              crc32  Calculate CRC32 using the polynomial from IEEE-802.3 (Ethernet).

              crc64  Calculate CRC64 using the polynomial from ECMA-182.  This is the default, since it is slightly better than CRC32 at detecting damaged files and the speed difference
                     is negligible.

              sha256 Calculate SHA-256.  This is somewhat slower than CRC32 and CRC64.

              Integrity of the .xz headers is always verified with CRC32.  It is not possible to change or disable it.

One can check how it works as follows:

    $ date > date.txt
    $ xz date.txt
    $ file date.txt.xz       # check for file format / xz archive
    date.txt.xz: XZ compressed data
    $ xz -l date.txt.xz      # check if it contains a checksum
    Strms  Blocks   Compressed Uncompressed  Ratio  Check   Filename
        1       1         88 B         30 B  2.933  CRC64   date.txt.xz
    
    $ xz -t date.txt.xz      # test integrity checksum
    $ echo $?                # return value 0 means OK
    0
    $ cat date.txt.xz | sed "s/f/ /g" > date-corrupted.txt.xz  # corrupt the archive by replacing some character (this wont work the same way every time)
    $ xz -l date-corrupted.txt.xz
    Strms  Blocks   Compressed Uncompressed  Ratio  Check   Filename
        1       1         88 B         30 B  2.933  CRC64   date-corrupted.txt.xz
    $ xz -t date-corrupted.txt.xz
    xz: date-corrupted.txt.xz: Compressed data is corrupt
    $ cat date.txt.xz | sed "s/f/ /g" > date-corrupted.txt.xz  # be aware that even "xz -l" can fail on a damaged archive
    $ xz -l date-corrupted.txt.xz
    xz: date-corrupted.txt.xz: Compressed data is corrupt
    $

So TL;DR version: default archive does contain a checksum that can be verified. It is possible to additionally provide explicit `sha256sum` file, but its usage will be bit less useful.

**Comment by BryanQuigley** [comment]
So should we drop the md5sum if we can verify it without it?

**Comment by BryanQuigley** [comment]
forgot this issue existed - proposed this https://github.com/sosreport/sos/pull/1441

**Comment by bmr-cymru** [comment]
I think even `md5` is a bit stronger than `crc64` (not sure of the details, but `crc64` is limited to detecting burst errors up to either 63 or 64 bits in length).

Given this is highly user visible, and potentially impacts user experience I think we need to coordinate very carefully with downstream consumers before changing any defaults.

**Comment by BryanQuigley** [comment]
I think 4.0 is the right time to make a change in this space.

I think it's important to not be using md5 at this point, so either:
 - just say crc64 is good enough
 - enable sha256sum as the default

**Comment by TurboTurtle** [comment]
So this missed the 4.0 boat, but we should probably make a decision here for either cutting over in 4.1 or pushing this way further out.

Since `sos-collector` is now integrated with `sos`, we have less extraneous bits to worry about.

@BryanQuigley @slashdd thoughts from the Ubuntu side on this going into 4.1 vs. later? I'll follow up with our RH QE folks internally.

**Comment by BryanQuigley** [comment]
People can always ignore the checksum, but if we are going to provide one it should be sha256 at this point.
